contrib/devtools/github-merge.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python2
+# Copyright (c) 2016 Bitcoin Core Developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+# This script will locally construct a merge commit for a pull request on a
+# github repository, inspect it, sign it and optionally push it.
+
+# The following temporary branches are created/overwritten and deleted:
+# * pull/$PULL/base (the current master we're merging onto)
+# * pull/$PULL/head (the current state of the remote pull request)
+# * pull/$PULL/merge (github's merge)
+# * pull/$PULL/local-merge (our merge)
+
+# In case of a clean merge that is accepted by the user, the local branch with
+# name $BRANCH is overwritten with the merged result, and optionally pushed.
+from __future__ import division,print_function,unicode_literals
+import os,sys
+from sys import stdin,stdout,stderr
+import argparse
+import subprocess
+
+# External tools (can be overridden using environment)
+GIT = os.getenv('GIT','git')
+BASH = os.getenv('BASH','bash')
+
+def git_config_get(option, default=None):
+    '''
+    Get named configuration option from git repository.
+    '''
+    try:
+        return subprocess.check_output([GIT,'config','--get',option]).rstrip()
+    except subprocess.CalledProcessError as e:
+        return default
+
+def retrieve_pr_title(repo,pull):
+    '''
+    Retrieve pull request title from github.
+    Return None if no title can be found, or an error happens.
+    '''
+    import urllib2,json
+    try:
+        req = urllib2.Request("https://api.github.com/repos/"+repo+"/pulls/"+pull)
+        result = urllib2.urlopen(req)
+        result = json.load(result)
+        return result['title']
```

Comment:
Could make sense to "sanitize" the title such that it fits in the git(hub) subject line.

That is, `return title[0:min(80, len(title)) - len("Merge pull request #xxxx: "]`